### ISSUE TITLE
Disable checkboxes for single items

### ISSUE THREAD

**Issue by AdityaAnand1** [spacecowboy/NoNonsense-FilePicker#142]
The [material design guidelines](https://material.io/guidelines/components/selection-controls.html) specifically state that checkboxes are to be used for multiple selection only. 

It looks a bit confusing since I'm sending off the user to pick one file and they see checkboxes which sort of interrupts "the flow", the smooth user experience that I'm aiming for.

Is there a way to simply hide them?

**Comment by spacecowboy** [comment]
Just use a custom layout:
http://spacecowboy.github.io/NoNonsense-FilePicker/example/custom_item_layout/

Den 3 maj 2017 13:00 skrev "Aditya Anand" <notifications@github.com>:

**Comment by AdityaAnand1** [comment]
Ok, so here's what I've tried:

* I used a custom layout: 

```
case LogicHandler.VIEWTYPE_CHECKABLE:
                v = LayoutInflater.from(getActivity()).inflate(R.layout.without_checkbox,
                        parent, false);
```

And I got a crash since something was expecting the `CheckableViewHolder` to have a checkbox.

* I tried setting it's visibility to invisible in my layout, but the visibility is reset in the CheckableViewHolder.

* So I just manually
```
case LogicHandler.VIEWTYPE_CHECKABLE:
                v = LayoutInflater.from(getActivity()).inflate(R.layout.nnf_filepicker_listitem_checkable,
                        parent, false);
                CheckableViewHolder holder = new CheckableViewHolder(v);
                holder.checkbox.setVisibility(View.INVISIBLE);
```

And that seems to get me what I want visually (although this doesn't need a custom layout, I can just reuse the ones in the library).

But the behavior I want is the user taps a file and it calls the "Ok" button action. What I did above just hid the checkbox. It's still there, so tapping it just does nothing as the filepicker only returns on clicking "Ok". Except now there's no indication as to which file is "selected" to be returned. I want it to return on the tap on the file item itself.

So, is there a way to hook in the upstream "Ok" action to the item's onclick? The user taps a file, I get the result in my activity.
